go/x509/verify_test.go

```diff
@@ -83,6 +84,30 @@ var verifyTests = []verifyTest{
 		errorCallback: expectExpired,
 	},
 	{
+		leaf:              googleLeaf,
+		intermediates:     []string{thawteIntermediate},
+		roots:             []string{verisignRoot},
+		currentTime:       1,
+		dnsName:           "www.google.com",
+		disableTimeChecks: true,
+
+		expectedChains: [][]string{
+			{"Google", "Thawte", "VeriSign"},
+		},
+	},
+	{
+		leaf:              googleLeaf,
+		intermediates:     []string{thawteIntermediate},
+		roots:             []string{verisignRoot},
+		currentTime:       10000000000,
+		dnsName:           "www.google.com",
+		disableTimeChecks: true,
+
+		expectedChains: [][]string{
+			{"Google", "Thawte", "VeriSign"},
```

Comment:
As I understand it, the expired/not-yet-valid cert is simulated by changing the value of currentTime so that the cert appears to be expired/not-yet-valid.  With this in mind I have just added a test and restructured them so that it is clear that a test that failed because of its validity period then passes if disableTimeChecks is true, for a cert that is expired, and a cert that is not-yet-valid against the currentTime.